cmd/storj-sim/console.go

```diff
@@ -0,0 +1,227 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/satellite/console"
+)
+
+type graphqlResponse struct {
+	Data   []byte
+	Errors []interface{}
+}
+
+func graphqlDo(client *http.Client, request *http.Request) (graphqlResponse, error) {
+	resp, err := client.Do(request)
+	if err != nil {
+		return graphqlResponse{}, err
+	}
+	defer func() {
+		err = resp.Body.Close()
+	}()
+
+	b, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return graphqlResponse{}, err
+	}
+
+	fmt.Println(string(b))
+
+	var response struct {
+		Data   json.RawMessage
+		Errors []interface{}
+	}
+
+	err = json.NewDecoder(bytes.NewReader(b)).Decode(&response)
+	if err != nil {
+		return graphqlResponse{}, err
+	}
+
+	return graphqlResponse{
+		Data:   []byte(response.Data),
+		Errors: response.Errors,
+	}, nil
+}
+
+func addExampleProjectWithKey(key *string, address string) error {
+	client := http.Client{}
+
+	var (
+		createUserFormat    = "mutation {createUser(input:{email:\"%s\",password:\"%s\",firstName:\"%s\",lastName:\"%s\"})}"
+		createProjectFormat = "mutation {createProject(input:{name:\"%s\",description:\"%s\"}){id}}"
+		createAPIKeyFormat  = "mutation {createAPIKey(projectID:\"%s\",name:\"%s\"){key}}"
+		tokenFormat         = "query {token(email:\"%s\",password:\"%s\"){token}}"
+	)
+
+	alice := console.CreateUser{
```

Comment:
Not quite sure why these types are necessary here? I would probably hardcode the full queries everywhere. This is a temporary solution.

Reply:
Just to make code more clear, i guess